price/pricing/pricer.go

```diff
@@ -0,0 +1,228 @@
+package pricing
+
+import (
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/fln/pprotect"
+
+	"github.com/mysteriumnetwork/payments/crypto"
+	"github.com/rs/zerolog/log"
+)
+
+type Bound struct {
+	Min, Max float64
+}
+
+type FiatPriceAPI interface {
+	MystUSD() float64
+}
+
+type Pricer struct {
+	priceAPI      FiatPriceAPI
+	priceLifetime time.Duration
+	mystBound     Bound
+
+	lock        sync.Mutex
+	lp          LatestPrices
+	cfgProvider ConfigProvider
+}
+
+func NewPricer(cfgProvider ConfigProvider, priceAPI FiatPriceAPI, priceLifetime time.Duration, sensibleMystBound Bound) (*Pricer, error) {
+	pricer := &Pricer{
+		cfgProvider:   cfgProvider,
+		priceAPI:      priceAPI,
+		priceLifetime: priceLifetime,
+		mystBound:     sensibleMystBound,
+	}
+
+	go schedulePriceUpdate(priceLifetime, pricer)
```

Comment:
I would probably go with the good old `Run()` `Stop()` but i guess this works too. 🤷🏻